### ISSUE TITLE
[nuxt3][vercel] Feature request: Support vercel provider for static app (nuxt generate)

### ISSUE THREAD

**Issue by hartmut-co-uk** [nuxt/image#617]
## Feature

Feature request: Support (nuxt3) nuxt/image with vercel provider for static app -> `nuxt generate`.

### Expected

Vercel config is generated and contains the "images" property which configures the Image Optimization feature. 

### Actual 

Website is generated, `<nuxt-image>` components are processed fine, but vercel config does not contain the "images" property.

### Reproduction example

https://github.com/hartmut-co-uk/nuxt3-generate-static-vercel-image

**Comment by mrdannyjohnson** [comment]
@hartmut-co-uk  This may be happening because static image generation is not supported yet (4th checkbox):
https://github.com/nuxt/image/discussions/548

**Comment by hartmut-co-uk** [comment]
I'd guess this work item would mean image variants (srcset) are actually generated + packaged at build ('generate') time using e.x. [IPX](https://v1.image.nuxtjs.org/providers/ipx/)?

Basically I might even prefer this option - but as far as this issue is concerned the [Vercel Image Optimization](https://vercel.com/docs/concepts/image-optimization#)(https://vercel.com/docs/concepts/image-optimization) should be used.
Basically the website is static, pre-rendered - but using vercel images, like when running on nitro, SSR..
(-> https://nuxt3-generate-static-vercel-image-lb046wh8w-info301.vercel.app/)

**Comment by danielroe** [comment]
Have you tried setting `provider: 'vercel'` in your image config?

**Comment by hartmut-co-uk** [comment]
hah, and I thought something was different from my actual project...

I've created following branch to show the state with `provider: 'vercel'` - the generated image src/srcset now show correct use of vercel path... but what's missing as stated in initial post seems to be   
->
> Vercel config is generated and contains the "images" property

**Comment by hartmut-co-uk** [comment]
any pointers which repo/package this actually would need to be solved with?
- https://github.com/nuxt/framework ?
- https://github.com/unjs/nitro ?

**Comment by hartmut-co-uk** [comment]
The page in question has only a hand full of routes, as some sort of workaround, I was able to solve what I was trying to achieve by simply configuring all my routes for prerender:
 
```ts
    nitro: {
        prerender: {
            routes: ['/', '/a', '/b', '/c']
        }
    },
```

It's deployed to vercel and not served via nitro functions but static as expected. Vercel image provider is still getting configured and working.

ℹ️  I'll keep the issue open, since I think it would still be good to have vercel image provider working with `nuxt generate`.

**Comment by Baroshem** [comment]
Closing as the fix was introduced in a linked PR.

Feel free to reopen if needed :)
